### ISSUE TITLE
Removing dead code

### ISSUE THREAD

**Issue by jstolarek** [links-lang/links#70]
In several places in the source code I have stumbled upon fragments of code that are commented out. This makes the code a lot less pleasant to read. `git blame` tells me that many of these fragments have been in commented out many years. This leads me to suspect that the code inside the comments might have bitrotten and will never be resurrected. I would like to remove that code. Are there any objections to doing so?

**Comment by jamescheney** [comment]
Not from me, but can I suggest doing this as a short-lived branch and pull request, so that anyone who thinks something should be retained has an opportunity to review / object?

--James

**Comment by jstolarek** [comment]
> can I suggest doing this as a short-lived branch and pull request, so that anyone who thinks something should be retained has an opportunity to review / object?

Yes, this is exactly my intention. I am also not planning to clean the whole source code in one go. Way too much work. I'll just focus on code fragments that I'm working on at the moment.

**Comment by jstolarek** [comment]
Here's a patch that removes dead code mainly from `types.ml` and `typeSugar.ml` (and some trailing whitespaces too). About 90% of deleted code are things imported in 2009 from SVN repo. I did not remove all of code that was commented out. More specifically, there is some code from 2014 that seems to be related to session types. I decided that this will be cut-off point. I left all of code from 2014 and never, everything older was deleted.
